content/scripts/build.js

```diff
@@ -982,7 +982,18 @@ class Builder {
     const rawHtml = fs.readFileSync(path.join(folder, "index.html"), "utf8");
     hasher.update(rawHtml);
 
-    const [renderedHtml, errors] = await this.renderHtml(rawHtml, metadata);
+    let renderedHtml;
+
+    // When 'source.htmlAlreadyRendered' is true, it simply means that the 'index.html'
+    // is already fully rendered HTML.
+    if (source.htmlAlreadyRendered) {
+      renderedHtml = rawHtml;
+    } else {
+      // XXX The this.renderHtml() method actually returns a tuple
+      // of (renderedHtml, errors) but we're not doing anything with the
+      // errors yet. Which we *will* do eventually.
+      [renderedHtml] = await this.renderHtml(rawHtml, metadata);
```

Comment:
Perhaps I just need to get mode JS into my bloodstream but I look at that syntax I have to think hard to appreciate what it does.

Reply:
Good point @peterbe. It's not obvious to me either, so perhaps your original way is better since it's clear and there's no wondering about it? Thanks for trying to placate me (😄 ) but feel free to revert back to your original code if you prefer that!